modules/ROOT/pages/indexes/semantic-indexes/vector-indexes.adoc

```diff
@@ -117,12 +119,56 @@ You can also create a vector index for relationships with a particular type on a
 ----
 CREATE VECTOR INDEX name IF NOT EXISTS
 FOR ()-[r:REL_TYPE]-() ON (r.embedding)
-OPTIONS {indexConfig: {
+OPTIONS { indexConfig: {
  `vector.dimensions`: $dimension,
  `vector.similarity_function`: $similarityFunction
 }}
 ----
 
+[[configuration-settings]]
+=== Configuration settings
+
+`vector.dimensions` ::
+The dimensionality of the vectors to be indexed, see xref:indexes/semantic-indexes/vector-indexes.adoc#embeddings[].
+Setting this value will ensure only vectors with the configured dimensionality will be indexed, and querying the index with a vector of a different dimensionality will return an error. It is recommended to provide the dimensions.
+Accepted values::: `INTEGER` between `1` and `4096` inclusively.
+Default Value::: None, though the setting can be omitted. The setting was mandatory prior to Neo4j 5.23.
+
+`vector.similarity_function` ::
+The name of the similarity function to use to define how similar two vectors are.
+To read more about the available similarity functions, see xref:indexes/semantic-indexes/vector-indexes.adoc#similarity-functions[].
+
+Accepted values::: `STRING`: `'cosine'`, `'euclidean'`.
+Default Value::: `'cosine'`. The setting was mandatory prior to Neo4j 5.23.
+
+`vector.quantization.enabled` label:new[Introduced in 5.23] ::
+Quantization is a technique to reduce the size of vector representations.
+Enabling quantization can accelerate search performance but can slightly decrease accuracy. On machines with limited memory, this is a recommended to always be enabled.
+Vector indexes created prior to Neo4j 5.23 have this setting effectively set as `false`.
+
+Accepted values::: `BOOLEAN`: `true`, `false`.
+Default value::: `true`
+
+==== Advanced configuration settings
+
+`vector.hnsw.m` label:new[Introduced in 5.23] ::
+The `M` parameter controls the maximum number of connections each node has in the HNSW graph.
+Increasing this value may lead to greater accuracy at the expense of increased index population and update times, especially for vectors with high dimensionality.
+Vector indexes created prior to Neo4j 5.23 have this setting effectively set as `16`.
+
+Accepted values::: `INTEGER` between `1` and `512` inclusively.
+Default value::: `16`
+
+`vector.hnsw.ef_construction` label:new[Introduced in 5.23] ::
+The number of nearest neighbors tracked during the insertion of vectors into the HNSW graph.
+Increasing this value increases the quality of the index, and may lead to greater accuracy (with diminishing returns) at the expense of increased index population and update times.
+Vector indexes created prior to Neo4j 5.23 have this setting effectively set as `100`.
+
+Accepted values::: `INTEGER` between `1` and `3200` inclusively.
+Default value::: `100`
+
+For differences between `vector.dimensions` and `vector.similarity_function` between vector index providers, see xref:indexes/semantic-indexes/vector-indexes.adoc#vector-index-providers[].
```

Comment:
Should this be here?

Reply:
moving a more generic version to the top of the section.